src/main/java/com/celements/store/CelHibernateStore.java

```diff
@@ -0,0 +1,101 @@
+package com.celements.store;
+
+import javax.inject.Singleton;
+
+import org.xwiki.component.annotation.Component;
+import org.xwiki.model.reference.DocumentReference;
+import org.xwiki.model.reference.ImmutableDocumentReference;
+
+import com.google.common.collect.Iterables;
+import com.xpn.xwiki.XWikiContext;
+import com.xpn.xwiki.XWikiException;
+import com.xpn.xwiki.doc.XWikiDocument;
+import com.xpn.xwiki.objects.BaseObject;
+import com.xpn.xwiki.store.XWikiHibernateStore;
+
+@Singleton
+@Component
+public class CelHibernateStore extends XWikiHibernateStore {
+
+  @Override
+  public void saveXWikiDoc(XWikiDocument doc, final XWikiContext context,
+      final boolean bTransaction) throws XWikiException {
+    // XWikiHibernateStore.saveXWikiDoc requires a mutable docRef
+    DocRefMutabilityExecutor<Void> exec = new DocRefMutabilityExecutor<Void>() {
+
+      @Override
+      public Void call(XWikiDocument doc) throws XWikiException {
+        CelHibernateStore.super.saveXWikiDoc(doc, context, bTransaction);
+        return null;
+      }
+    };
+    exec.execute(doc);
+  }
+
+  @Override
+  public XWikiDocument loadXWikiDoc(XWikiDocument doc, final XWikiContext context)
+      throws XWikiException {
+    // XWikiHibernateStore.loadXWikiDoc requires a mutable docRef
+    DocRefMutabilityExecutor<XWikiDocument> exec = new DocRefMutabilityExecutor<XWikiDocument>() {
+
+      @Override
+      public XWikiDocument call(XWikiDocument doc) throws XWikiException {
+        return CelHibernateStore.super.loadXWikiDoc(doc, context);

```

Comment:
The docRefs contained in objects are created over BaseObject.setName and BaseObject.setClassName. To solve this problem we would need to replace the DocumentReferenceResolver implementations with implementations returning ImmutableDocumentReferences.

Nevertheless please create the two issues for inlining the methods and improving load performance afterwards. Moreover I think on line 750-754 objects could get lost on loading! Please mention on the issue. Mention the issue numbers in CelHibernateStore. I will then approve :-)